### PR TITLE
02 movement system

### PR DESCRIPTION
**Pull request by SB-Lamia**
Quite Abit has been changed and reworked... I think its better if I sit down in person with you guys thursday....

First the arrows.
image
image

Each arrow has a arrow 3d model to point in the direction. and a text car above it giving a brief overview of where its heading.
Each arrow has a scripts which holds 
The location its going.
The text on the arrow.
The rotation of the character after you teleport.
The parent of the arrow.
NOTE: that each location has a set of arrows you can tap to go to the next location... but I did not want arrows randomly everywhere. So after you teleport, it turns on the new locations arrows and turns off the old locations arrows.

Next I spent 5 mins on a really quick movement system for the camera. Because I couldnt look around to see the arrows for testing...

This also meant I had to detach the camera from the character prefab and make it a child. Doesnt seem to be broken. Also the pc movement script HAS been disabled. Because of the reason it was preventing my character from rotating. I do not know why, but it was.

Lastly I moved the door in one of the rooms because it was pissing me off when I was trying to find a spot for my character... this is the door as it is. does it really look that bad?

image

Next I added a touch system. No mouse has not been added. but touch has been added. You can tap the arrows and green info cards. Which are all handled by this single script. I will comment it to explain but it manages alot of the tapping. it is attached to the Character script. 

Lastly there is a few bugs.
- The text on the arrows and the arrows themself do not look that nice... I assumed we could spend a future update making them look nicer
- If you start your tap hold on a intractable (something with a collider (the arrows or text)) then it will interact with that first. I have yet to find a way to update it so that it will consider the hold tap first before the quick tap and release. 

image

Any questions please ask. Cause like this is beyond complication but still really as simple as it gets. Since if you think about it... this one script is managing the teleporting around locations, the movement of the camera, and the popup trivia about the toilets.

### CODE REVIEW

**Review by JadenCooper — CHANGES_REQUESTED** [review]
The Movement Is Nice And I Don't Notice Any Drop From The Gravity. I Think We Don't Need The Text Above The Arrows To Tell The User Where They Go, The Arrows Themselves Should Be Good Enough. The Arrows Should Be Turned Into A Prefab As They Are A Repeating Game Object. The Colliders Should Be Shrunk As It Was Too Easy To Keep Teleporting By Accident. I Believe The Look System Needs To Be Moved Over To The New Input System So The Checks Can Be Taken Out Of The Update So It You Don't End Up With This.
image
Once We Merge It All Together I Think We Should Make The Arrow Inherit From The Interactable Script In The Display Text PR So That We Would Not Need To Chain If Statements To Check What Kind Of Object We re Trying To Click On

File: Assets/Scripts/LookingMobile.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        }
+        else if (Input.touchCount > 0 && Input.touches[0].phase == TouchPhase.Began)
+        {
+            Ray raycast = Camera.main.ScreenPointToRay(Input.GetTouch(0).position);
+            RaycastHit raycastHit;
+            if (Physics.Raycast(raycast, out raycastHit))
+            {
+                if (raycastHit.collider.CompareTag("Arrow"))
+                {
+                    ArrowMovement objectHit = raycastHit.collider.gameObject.transform.GetComponent<ArrowMovement>();
```
Comment (JadenCooper) [inline Assets/Scripts/LookingMobile.cs:36]: This Stuff Should Be Moved Over The Arrow Script So Its OO

Comment (SB-Lamia) [reply Assets/Scripts/LookingMobile.cs:36]: It shouldnt for a few reasons. 

One the arrow script is attached to 20-30 game objects across the map. That would mean each arrow is checking in their update for a raycast hit. and that is alot of unnecessary duplication of code. The arrow script only needs to know the information. it doesnt need to actually teleport or check.

2nd the looking movement already uses the same touch system. and so if you were to put it in another script you would have multiple raycasts out. This script is not managing the arrows. It is simply checking if you have interacted with an arrow. Then teleports you to the location that arrow has in its script.

(Also I forgot to push a few last changes (like the comments you saw above. so am gonna push that now)

**Comment by SB-Lamia** [comment]
> The Movement Is Nice And I Don't Notice Any Drop From The Gravity. I Think We Don't Need The Text Above The Arrows To Tell The User Where They Go, The Arrows Themselves Should Be Good Enough. The Arrows Should Be Turned Into A Prefab As They Are A Repeating Game Object. The Colliders Should Be Shrunk As It Was Too Easy To Keep Teleporting By Accident. I Believe The Look System Needs To Be Moved Over To The New Input System So The Checks Can Be Taken Out Of The Update So It You Don't End Up With This. image Once We Merge It All Together I Think We Should Make The Arrow Inherit From The Interactable Script In The Display Text PR So That We Would Not Need To Chain If Statements To Check What Kind Of Object We re Trying To Click On

I attempted the new movement system. Though I found it was overly complicated for what we were wanting to achieve. Would it work better? posibly yes. would it takes hours alone to setup? also yes. I spent 5 mins on this looking system and achieved a very usuable result. I dont see the point in wasting time to get the new movement system working for this sprint. It can be looked into for a future sprint. But for the moment I dont believe its nesscary.

I am happy to remove the text or even fix it up a little bit. I thought it was nice to show the user male vs female. but yeah the text could easily be removed and im fine with it.

And yeah I should of made it a prefab. I can rework them into prefabs. Same with the colliders. I am happy to shrink them down abit. 

And I dont believe you can make it a child of interactable. or it would really do anything. because the if and else ifs are very important. It has to check and they do very different things depending on which are hit. It would only work if everything leading up to it was the same and then the thing being excuted was different. Again we can try though I dont believe it would work any better then what is currently there since there is really not much of any repeating code. it looks similar, but each thing is doing something very different.

**Comment by JadenCooper** [comment]
What I Mean With The Intractable Script Is That Arrow Or Any Other Intractable Would Inherit It Like This 
image
Which Is Just This
image
So We Can Do This
image

**Comment by JadenCooper** [comment]
The Public Override Would Be In Say The Arrow Script But Would Do Its Own Unique Stuff Compared To Any Other Intractables It Just Gets Rid Of All The If Statements On The Click/Press To Figure Out What It Is

**Comment by JadenCooper** [comment]
But This Is Stuff We Will Have To Do Once  Its All Merged

**Review by Sharkules — APPROVED** [review]
That enormous comment chain covers everything I would have said! Looks like we have a handful of things to look at tweaking next sprint. 

Other than that, looks fantastic!

**Review by JadenCooper — APPROVED** [review]

**Pull request merged**